File: AquaTrack/src/main/java/com/aquatrack/service/FloorService.java
package com.aquatrack.service;

import com.aquatrack.dto.floor.FloorRequestDto;
import com.aquatrack.dto.floor.FloorResponseDto;

import java.util.List;

/**
 * Service interface for Floor management.
 */
public interface FloorService {

    /**
     * Creates a new floor for a building.
     *
     * @param buildingId Building ID
     * @param requestDto Floor request data
     * @return Created floor details
     */
    FloorResponseDto createFloor(Long buildingId,
                                 FloorRequestDto requestDto);

    /**
     * Returns all floors of a building.
     *
     * @param buildingId Building ID
     * @return List of floors
     */
    List<FloorResponseDto> getAllFloors(Long buildingId);

    /**
     * Returns floor details.
     *
     * @param buildingId Building ID
     * @param floorId Floor ID
     * @return Floor details
     */
    FloorResponseDto getFloorById(Long buildingId,
                                  Long floorId);

    /**
     * Updates floor details.
     *
     * @param buildingId Building ID
     * @param floorId Floor ID
     * @param requestDto Updated floor data
     * @return Updated floor details
     */
    FloorResponseDto updateFloor(Long buildingId,
                                 Long floorId,
                                 FloorRequestDto requestDto);

    /**
     * Deletes a floor.
     *
     * @param buildingId Building ID
     * @param floorId Floor ID
     */
    void deleteFloor(Long buildingId,
                     Long floorId);

    /**
     * Returns all floors of a building assigned to the
     * currently logged-in Manager.
     *
     * Access is granted only if the manager is assigned
     * to the specified building.
     *
     * @param buildingId Building ID
     * @return List of assigned floors
     */
    List<FloorResponseDto> getManagerFloors(Long buildingId);

}

File: Aquatrack-frontend/src/services/floorService.js
import api from "../api/axios";

// ==========================================
// Get All Floors
// ==========================================

export async function getFloors(buildingId) {

    const response = await api.get(
        `/buildings/${buildingId}/floors`
    );

    return response.data;

}

// ==========================================
// Get Floor By Id
// ==========================================

export async function getFloorById(
    buildingId,
    floorId
) {

    const response = await api.get(
        `/buildings/${buildingId}/floors/${floorId}`
    );

    return response.data;

}

// ==========================================
// Create Floor
// ==========================================

export async function createFloor(
    buildingId,
    data
) {

    const response = await api.post(
        `/buildings/${buildingId}/floors`,
        data
    );

    return response.data;

}



// ==========================================
// Update Floor
// ==========================================

export async function updateFloor(
    buildingId,
    floorId,
    data
) {

    const response = await api.put(
        `/buildings/${buildingId}/floors/${floorId}`,
        data
    );

    return response.data;

}

// ==========================================
// Delete Floor
// ==========================================

export async function deleteFloor(
    buildingId,
    floorId
) {

    const response = await api.delete(
        `/buildings/${buildingId}/floors/${floorId}`
    );

    return response.data;

}
